### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by haunv3**

### CODE REVIEW

File: src/hooks/useMetamask.ts [hunk 1]
```diff
@@ -10,20 +10,25 @@ export const injected = new InjectedConnector({
...
   const web3React = useWeb3React();
   const { pathname } = useLocation();
   const [chainInfo] = useGlobalState('chainInfo');
   const [metamaskAddress, setMetamaskAddress] =
     useGlobalState('metamaskAddress');
 
   useEffect(() => {
+    if(isInterval) return;
     eagerConnectBsc();
   }, [pathname]);
```
Comment (tubackkhoa) [inline src/hooks/useMetamask.ts:23]:
```js
useEffect(() => {
    if(isInterval) return;
    eagerConnectBsc();
  }, [pathname, isInterval]);
```

File: src/hooks/useMetamask.ts [hunk 2]
```diff
@@ -10,20 +10,25 @@ export const injected = new InjectedConnector({
...
 
   const eagerConnectBsc = () => {
```
Comment (tubackkhoa) [inline src/hooks/useMetamask.ts:25]: should change to **eagerConnectEvm** then check chainId is default Bsc or ethereum

File: src/rest/api.ts [hunk 3]
```diff
@@ -245,14 +245,29 @@ async function fetchNativeTokenBalance(
   lcd?: string,
   shouldKeepOriginal?: boolean
 ): Promise<number | string> {
-  const url = `${lcd ?? network.lcd
+  let lcdFetchTokenBalance = lcd ?? network.lcd;
```
Comment (tubackkhoa) [inline src/rest/api.ts:248]: no need, lcd is load balancing at nginx

File: src/rest/api.ts [hunk 4]
```diff
@@ -245,14 +245,29 @@ async function fetchNativeTokenBalance(
...
+  }
+  const url = `${lcdFetchTokenBalance
     }/cosmos/bank/v1beta1/balances/${walletAddr}/by_denom?denom=${denom}`;
   const res: any = (await axios.get(url)).data;
   const amount = res.balance.amount;
   if (shouldKeepOriginal) return amount;
   return parseInt(amount);
 }
 
+const pingUrl = async (url: string) => {
```
Comment (tubackkhoa) [inline src/rest/api.ts:261]: fallback after execution, not ping

**Review by tubackkhoa — APPROVED** [review]

**Review by phutxorai — APPROVED** [review]

**Review by im6h — APPROVED** [review]

**Pull request merged**